dbprovider/dbprovider.go

```diff
@@ -0,0 +1,12 @@
+package dbprovider
+
+import (
+	"github.com/sorenmat/k8s-rds/crd"
+	"k8s.io/client-go/kubernetes"
+)
+
+type DBProvider interface {
+	CreateDatabase(kubectl *kubernetes.Clientset, publicAccessible bool, password string, db *crd.Database) (string,error)
```

Comment:
publicAccessible bool and password string should be available from the crd.Database